src/coreclr/jit/morph.cpp

```diff
@@ -14544,8 +14544,10 @@ void Compiler::fgExpandQmarkStmt(BasicBlock* block, Statement* stmt)
     // if they are going to be cleared by fgSplitBlockAfterStatement(). We currently only do this only
     // for the GC safe point bit, the logic being that if 'block' was marked gcsafe, then surely
     // remainderBlock will still be GC safe.
-    BasicBlockFlags propagateFlags = block->bbFlags & BBF_GC_SAFE_POINT;
-    BasicBlock*     remainderBlock = fgSplitBlockAfterStatement(block, stmt);
+    BasicBlockFlags propagateFlagsToRemainder = block->bbFlags & BBF_GC_SAFE_POINT;
+    // Conservatively mark all blocks as having IDX_LEN since we don't know where exactly it went
+    BasicBlockFlags propagateFlagsToAll = block->bbFlags & BBF_HAS_IDX_LEN;
```

Comment:
This should use `BBF_COPY_PROPAGATE`

Reply:
> This should use `BBF_COPY_PROPAGATE`

That will decrease throughput at this moment for no benefit - e.g. we'll mark 4 blocks with "has nullcheck" and will have to visit them all in earlyprop. Ideally, QmarkGenTree should store BB flags it needs to propagate IMO. I propose we add flags on demand here since we use QMARK only in a few specific cases (none of them need to propagate flags atm)

Reply:
>  can you add asserts that check for GT_NULLCHECK, GT_MDARR_LENGTH, GT_ALLOCOBJ at least then (can use gtTreeContainsOper for this)?

It will assert anyway if we have a block without e.g. HAS_NULLCHECK with an actuall nullcheck so no need in an extra assert - so if someone adds a new kind of qmark where it's needed (prior my PR it was never needed) they will hit it. The extra regression from `BBF_COPY_PROPAGATE` is another +0.02% or so

Reply:
Ok, will leave it up to you -- although I don't think 0.02% is that much to have this be always correct

Reply:
It's not urgent, let's have a CI run with BBF_COPY_PROPAGATE